arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -659,6 +678,73 @@ RestStatus RestCursorHandler::createQueryCursor() {
   return registerQueryOrCursor(body);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief shows the batch given by <batch-id> if it's the last cached batch
+/// response on a retry, and does't advance cursor
+////////////////////////////////////////////////////////////////////////////////
+
+RestStatus RestCursorHandler::showLastBatchRetry() {
+  std::vector<std::string> const& suffixes = _request->suffixes();
+
+  if (suffixes.size() != 2) {
+    generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "expecting POST /_api/cursor/<cursor-id>/<batch-id>");
+    return RestStatus::DONE;
+  }
+
+  std::string const& id = suffixes[0];
+
+  auto cursors = _vocbase.cursorRepository();
+  TRI_ASSERT(cursors != nullptr);
+
+  auto cursorId = static_cast<arangodb::CursorId>(
+      arangodb::basics::StringUtils::uint64(id));
+  bool busy;
+  _cursor = cursors->find(cursorId, busy);
+
+  if (_cursor == nullptr) {
+    if (busy) {
+      generateError(GeneralResponse::responseCode(TRI_ERROR_CURSOR_BUSY),
+                    TRI_ERROR_CURSOR_BUSY);
+    } else {
+      generateError(GeneralResponse::responseCode(TRI_ERROR_CURSOR_NOT_FOUND),
+                    TRI_ERROR_CURSOR_NOT_FOUND);
+    }
+    return RestStatus::DONE;
+  }
+  if (!_cursor->isRetriable()) {
+    generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_BAD_PARAMETER,
+                  "expecting retriable option to be true");
+    return RestStatus::DONE;
+  }
+
+  _cursor->setWakeupHandler(withLogContext(
+      [self = shared_from_this()]() { return self->wakeupHandler(); }));
+
+  TRI_ASSERT(_cursor != nullptr);
```

Comment:
I think this assert can be removed. The pointer was already dereferenced twice before it, so the assert is of no value.